roles/variables/vars/main.yml

```diff
@@ -138,33 +307,29 @@ kafka_broker_properties:
+      controller.quorum.voters: "{% for inventory_hostname in groups.kafka_controller|default([]) %}{% if loop.index > 1%},{% endif %}{{inventory_hostname[-1]|int + 9990}}@{{inventory_hostname}}:{{ kafka_controller_listeners['controller']['port'] }}{%endfor%}"
       controller.listener.names: "{{kafka_controller_listeners['controller']['name']}}"
       listener.security.protocol.map: "{% for listener in kafka_controller_listeners|dict2items %}{% if loop.index > 1%},{% endif %}{{ listener['value']['name'] }}:{{ listener['value'] | confluent.platform.kafka_protocol_defaults(kafka_controller_ssl_enabled, kafka_controller_sasl_protocol)}}{% endfor %},{% for listener in kafka_broker_listeners|dict2items %}{% if loop.index > 1%},{% endif %}{{ listener['value']['name'] }}:{{ listener['value'] | confluent.platform.kafka_protocol_defaults(ssl_enabled, sasl_protocol)}}{% endfor %}"
       listeners: "{% for listener in kafka_broker_listeners|dict2items %}{% if loop.index > 1%},{% endif %}{{ listener['value']['name'] }}://{{ listener['value']['ip'] | default('') }}:{{ listener['value']['port'] }}{% endfor %}"
       inter.broker.listener.name: "{{kafka_broker_listeners[kafka_broker_inter_broker_listener_name]['name']}}"
   broker_on_zookeeper:
-    enabled: "{{not kraft_enabled|bool and inventory_hostname in groups.kafka_broker}}"
+    enabled: "{{not kraft_enabled|bool}}"
     properties:
       advertised.listeners: "{% for listener in kafka_broker_listeners|dict2items %}{% if loop.index > 1%},{% endif %}{{ listener['value']['name'] }}://{{ listener['value']['hostname'] | default(hostvars[inventory_hostname]|confluent.platform.resolve_hostname)  }}:{{ listener['value']['port'] }}{% endfor %}"
       listeners: "{% for listener in kafka_broker_listeners|dict2items %}{% if loop.index > 1%},{% endif %}{{ listener['value']['name'] }}://{{ listener['value']['ip'] | default('') }}:{{ listener['value']['port'] }}{% endfor %}"
       listener.security.protocol.map: "{% for listener in kafka_broker_listeners|dict2items %}{% if loop.index > 1%},{% endif %}{{ listener['value']['name'] }}:{{ listener['value'] | confluent.platform.kafka_protocol_defaults(ssl_enabled, sasl_protocol)}}{% endfor %}"
       zookeeper.connection.timeout.ms: 18000
       zookeeper.connect: "{{ groups['zookeeper'] | default(['localhost']) | confluent.platform.resolve_hostnames(hostvars) | join(':' + zookeeper_client_port|string + ',') }}:{{zookeeper_client_port}}{{zookeeper_chroot}}"
       inter.broker.listener.name: "{{kafka_broker_listeners[kafka_broker_inter_broker_listener_name]['name']}}"
+  kraft_sasl_plain:
+    enabled: "{{kraft_enabled|bool and kafka_controller_sasl_protocol=='plain'}}"
+    properties:
+      listener.name.controller.plain.sasl.jaas.config: "{{plain_jaas_config}}"
+      sasl.mechanism.controller.protocol: PLAIN
```

Comment:
Why are we hardcoding this? shouldn't we use kafka_protocol_defaults to get the protocol?

Reply:
but this condition itself runs when we use plain protocol. So shouldn't this be enough?